src/pybind/CMakeLists.txt

```diff
@@ -67,6 +67,23 @@ if(WITH_MGR)
   elseif(NOT WITH_PYTHON3 AND MGR_PYTHON_VERSION_MAJOR EQUAL 3)
     message(FATAL_ERROR "mgr plugins require python3 binding")
   endif()
+
+  if (USE_OPENSSL)
+    execute_process(
+      COMMAND ${PYTHON${PYTHON_VERSION}_EXECUTABLE} -c "import ssl; print('.'.join(map(str,ssl.OPENSSL_VERSION_INFO[0:3])))"
+      RESULT_VARIABLE PYSSL_RESULT
+      OUTPUT_VARIABLE PYSSL_VER
+      ERROR_QUIET
+      )
+    if (NOT ${PYSSL_RESULT})
+      if (NOT (OPENSSL_VERSION VERSION_EQUAL PYSSL_VER))
+        message(FATAL_ERROR "Python and Ceph link to different openssl versions: ${PYSSL_VER} vs ${OPENSSL_VERSION}")
+      endif()
+    else()
+      message(WARNING "could not determine ssl version of python crypto lib")
```

Comment:
nit,

> could not determine the OpenSSL version used by python ssl module

also, could make this `FATAL`, as per https://docs.python.org/2/library/ssl.html and https://docs.python.org/3/library/ssl.html. we *should* always have zero retcode of the command line above.